### PR TITLE
general: fix models for PostgreSQL compatibility

### PR DESCRIPTION
**Pull request by hachreak**
* FIX Fixes models to permit the database creation with
  PostgreSQL server.

* COMPATIBILITY Enables 'sql_mode' as 'ansi_quotes' for quotes
  compatibility for MySQL.

Signed-off-by: Leonardo Rossi <leonardo.r@cern.ch>

### CODE REVIEW

File: invenio/legacy/dbquery_postgresql.py [hunk 1]
```diff
@@ -225,30 +225,9 @@ def get_table_update_time(tablename, run_on_slave=False):
...
-            # MySQL-4.1 and 5.0 have creation_time in 11th position,
-            # so return next column:
-            update_times.append(str(row[12]))
-        else:
-            # MySQL-4.0 has creation_time in 10th position, which is
-            # of type datetime.datetime or str (depending on the
-            # version of MySQLdb), so return next column:
-            update_times.append(str(row[11]))
-    return max(update_times)
+    # FIXME how can I implement it with PostgreSQL???
```
Comment (hachreak) [inline invenio/legacy/dbquery_postgresql.py]: @jirikuncar do you have some idea on how can I make an equivalent function for postgresql? T_T

File: invenio/modules/access/control.py [hunk 2]
```diff
@@ -238,11 +240,12 @@ def acc_add_role(name_role, description,
...
     """
 
-    if not run_sql("""SELECT name FROM accROLE WHERE name = %s""", (name_role, )):
-        res = run_sql("""INSERT INTO accROLE (name, description,
+    if not run_sql("""SELECT name FROM "accROLE" WHERE name = %s""", (name_role, )):
+        res = run_sql("""INSERT INTO "accROLE" (name, description,
                             firerole_def_ser, firerole_def_src)
                          VALUES (%s, %s, %s, %s)""",
-            (name_role, description, firerole_def_ser, firerole_def_src))
+            (name_role, description, bytearray(firerole_def_ser) if firerole_def_ser is not None else None, firerole_def_src))
```
Comment (hachreak) [inline invenio/modules/access/control.py]: @jirikuncar do you know some contraindications for the use of `bytearray()`?

**Comment by hachreak** [comment]
To work, it's needed to install postgresql contrib:
```bash
sudo apt-get install postgresql-contrib-9.3
```
and, run the `query`:
```sql
CREATE EXTENSION pgcrypto;
```

File: invenio/legacy/dbquery_mysql.py [hunk 3]
```diff
@@ -535,3 +536,28 @@ def real_escape_string(unescaped_string, run_on_slave=False):
     escaped_string = connection_object.escape_string(unescaped_string)
     return escaped_string
 
+
+def aes_encrypt(field, password):
```
Comment (jirikuncar) [inline invenio/legacy/dbquery_mysql.py]: @hachreak please check #2873 made by @lnielsen

Comment (hachreak) [inline invenio/legacy/dbquery_mysql.py]: The `aes_encrypt()` is only useful to make compatible the legacy code with postgresql until we'll finish to port all legacy to v2.0.
So, @lnielsen code is welcome! XD

**Comment by hachreak** [comment]
@jirikuncar I rebased and resolved conflicts with aes_encrypt! :)

File: invenio/ext/sqlalchemy/__init__.py [hunk 4]
```diff
@@ -100,17 +100,17 @@ def update_mutable_dict(self, *args, **kwargs):
...
 # def compile_text(element, compiler, **kw):
 #     """Redefine Text filed type for PostgreSQL."""
 #     return 'TEXT'
 
-
-# @compiles(types.VARBINARY, 'postgresql')
-# def compile_text(element, compiler, **kw):
-#     """Redefine VARBINARY filed type for PostgreSQL."""
-#     return 'BYTEA'
+@compiles(types.VARBINARY, 'postgresql')
```
Comment (jirikuncar) [inline invenio/ext/sqlalchemy/__init__.py]: it's failing on travis :(

**Comment by hachreak** [comment]
ping @jirikuncar :dancers:

**Pull request merged**